report object name upon mangle_docstrings exception

jnothman opened the pull request.

jorisvandenbossche commented:
Does this automatically integrate with sphinx logging? (that warnings are printed, and their verbose option). Maybe we can do `from sphinx.util import logging` instead of `import logging` (like eg in autodoc: https://github.com/sphinx-doc/sphinx/blob/master/sphinx/ext/autodoc/directive.py)

jorisvandenbossche commented on numpydoc/numpydoc.py:
```diff
@@ -124,13 +127,17 @@ def mangle_docstrings(app, what, name, obj, options, lines):
+                doc = unicode(doc)
+            lines[:] = doc.split(u_NL)
+        except:
+            logger.error('While processing numpydoc for %r', name)
```
Suggestion: add `[numpydoc] ` at the front of the message? That's also how sphinx extensions like autodoc/autosummary do it

jnothman commented:
done, though that prefixing is redundant with registering the logger to the
file

On 12 Apr 2018 9:34 pm, "Joris Van den Bossche" <notifications@github.com>
wrote:

> *@jorisvandenbossche* commented on this pull request.
> ------------------------------
>
> In numpydoc/numpydoc.py
> <https://github.com/numpy/numpydoc/pull/164#discussion_r181049566>:
>
> > -                             builder=app.builder)
> -        if sys.version_info[0] >= 3:
> -            doc = str(doc)
> -        else:
> -            doc = unicode(doc)
> -        lines[:] = doc.split(u_NL)
> +        try:
> +            doc = get_doc_object(obj, what, u_NL.join(lines), config=cfg,
> +                                 builder=app.builder)
> +            if sys.version_info[0] >= 3:
> +                doc = str(doc)
> +            else:
> +                doc = unicode(doc)
> +            lines[:] = doc.split(u_NL)
> +        except:
> +            logger.error('While processing numpydoc for %r', name)
>
> Suggestion: add [numpydoc] at the front of the message? That's also how
> sphinx extensions like autodoc/autosummary do it
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/numpy/numpydoc/pull/164#pullrequestreview-111581726>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEz67aRoIUL8UixRbs6O9mgyfZjgrrlks5tnzvSgaJpZM4TNYTY>
> .
>

jorisvandenbossche commented:
> done, though that prefixing is redundant with registering the logger to the file

Hmm, yes, it seems they only do the [] for debug messages (and some info), and use the location for warnings and errors (at least it looks not fully consistent for somebody who is not familiar with it)

rgommers commented:
`from sphinx.util import logging` is not present in Sphinx 1.2.3, which failed CI. I'd be fine with bumping the minimum Sphinx version.

rgommers commented:
I tested this locally, LGTM. +1 for merging when CI issue is resolved.

rgommers commented:
Updated the tested Sphinx versions from `latest, 1.2.3` to `latest, 1.6.5`. TravisCI is happy now, so merging.
